Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1924,18 +1924,21 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
                 try undoColl.deleteOne([idField: documentIdBeforeUpdate])
             }
 
-            deferredEventEmittingBlock = {
-                if triggerNamespace {
-                    self.triggerListening(to: namespace)
-                }
-                self.emitEvent(documentId: documentId, event: event);
-            }
+            self.emitEvent(documentId: documentId, event: event);
```

Comment:
[nit] this can be moved outside the critical section after the triggerListening call